### ISSUE TITLE
[Feature]

### ISSUE THREAD

**Issue by jwalatham** [wiedehopf/tar1090#306]
It would be great to utilise the Openlayers ruler code, to draw a line between two points on the map and measure the distance and bearing between them. For example, to measure how far an aircraft is from my location or another point, or the distance between two aircraft.

**Comment by RundesBalli** [comment]
Very nice idea, I support this feature request. :-)

**Comment by mpentler** [comment]
I've wanted to try and do this for a while but couldn't figure it out. It'd be like ATC staff have on their displays and would be cool for better situational awareness. I'd love to see this but foresee it being difficult to integrate with scrolling and map clicking. Maybe a button up top to enter that mode?

**Comment by mpentler** [comment]
Well, I started to learn how to make a layer and draw on it, and now I've added a ruler button that turns the layer on and off and draws a dot at 0,0.

Baby steps. This is helping me to learn the tar1090 flow too. I need to figure out where to add some click handlers to be able to draw on the map, though.

**Comment by wiedehopf** [comment]
Really one issue is where to put an extra button.

I suppose the extra button can be enabled from the settings menu ...

**Comment by mpentler** [comment]
You're right. I am not running globe mode on my local install, so I had no replay button and could stick it there. Otherwise it'd just have to go next to it, and then that's 5 buttons on the top row U/H/T/Replay/Ruler

Other option is a button on the top left, somewhere untouched by other things right now. Although my ruler controls are currently displayed up there when you turn the ruler mode on. Could make it a drop down thing - there's too many possibilities with this. I am trying to match the design of the map as-is right now.

**Comment by mpentler** [comment]
Got a working prototype going, but I don't think my code is really up to scratch. I will probably put in a branch on my own fork.

![image](https://github.com/wiedehopf/tar1090/assets/27921983/8bb06d57-4af0-42df-9c78-c3d105c39a75)
![image](https://github.com/wiedehopf/tar1090/assets/27921983/fa1dc849-545d-41d4-af71-27a81b071f3f)

**Comment by mpentler** [comment]
Now up on my repo, ruler branch. It's very shit.

**Comment by Telecomeansa** [comment]
Good morning, how are you? I'm interested in implementing distances between aircraft, please help me.
